quill-sql/src/main/scala/io/getquill/context/QueryExecution.scala

```diff
@@ -594,13 +595,16 @@ object PrepareDynamicExecution {
         case NotFound(uid)           => s"NotFoundPlanter($uid)"
       }
     }
-
     val sortedEncodeables =
       uidsOfScalarTags
         .map { uid =>
           encodeablesMap.get(uid) match {
-            case Some(element) => UidStatus.Primary(uid, element)
-            case None =>
+            case Some(head::Nil) =>
```

Comment:
Instead of having a `var` we should convert that to a flatMap, that way there a no hidden "iterations", the issue with flatmap is that we would end up with duplications inside the final `sortedEncodeables` which is not a good solution, and as far as I could tell order is important, so we can't use a `Set` inside of a `List`, however we could do the following:

```scala
    val encodeablesMap =
      lifts.groupBy(_.uid)

    val secondaryEncodeablesMap =
      secondaryLifts.groupBy(_.uid)

    val sortedEncodeables =
      uidsOfScalarTags.distinct
        .flatMap { uid =>
          encodeablesMap.get(uid) match {
            case Some(elements) => elements.map(UidStatus.Primary(uid, _))
            case None =>
              secondaryEncodeablesMap.get(uid) match {
                case Some(elements) => elements.map(UidStatus.Secondary(uid, _))
                case None          => UidStatus.NotFound(uid) :: Nil
              }
          }
        }
 ```